woodwork/logical_types.py

```diff
@@ -617,3 +604,36 @@ class PostalCode(LogicalType):
     "string",
     "timedelta64[ns]",
 }
+
+
+def _regex_validate(logical_type, series, return_invalid_values):
+    """Validates data values based on the logical type regex in the config."""
+    regex_key = {
+        "url": "url_inference_regex",
+        "email_address": "email_inference_regex",
+    }[logical_type.type_string]
+    regex = config.get_option(regex_key)
```

Comment:
I like the use of this `_regex_validate` function to clean up the individual logical type methods. 

I'm thinking we may want to keep the logic for getting the proper regex from the config inside the logical type, rather than putting logical type specific logic inside this function. That way, all the logical type specific stuff stays with the logical type and isn't split between the logical type method and this function.

Thoughts?

Reply:
sure I don't mind splitting